test/integration/certificates/metrics_controller_test.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificates
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	controllerpkg "github.com/jetstack/cert-manager/pkg/controller"
+	controllermetrics "github.com/jetstack/cert-manager/pkg/controller/certificates/metrics"
+	logf "github.com/jetstack/cert-manager/pkg/logs"
+	"github.com/jetstack/cert-manager/pkg/metrics"
+	"github.com/jetstack/cert-manager/test/integration/framework"
+	"github.com/jetstack/cert-manager/test/unit/gen"
+	testnet "github.com/jetstack/cert-manager/test/unit/net"
+)
+
+// TestMetricscontoller performs a basic test to ensure that Certificates
+// metrics are exposed when a Certificate is created, updated, and removed when
+// it is deleted.
+func TestMetricsController(t *testing.T) {
+	config, stopFn := framework.RunControlPlane(t)
+	defer stopFn()
+
+	// Build, instantiate and run the issuing controller.
+	_, factory, cmClient, cmFactory := framework.NewClients(t, config)
+
+	metricsPort, err := testnet.FreePort()
```

Comment:
Huh, how come [here](https://github.com/jetstack/cert-manager/pull/2923/files#diff-fbfc336e484fcf672a6cb108dabcf25cR63) we are able to bind to `:0`, but in this unit test we cannot?